employee-rostering-frontend/src/ui/pages/contract/ContractsPage.tsx

```diff
@@ -0,0 +1,140 @@
+  updateContract: typeof contractOperations.updateContract;
+  removeContract: typeof contractOperations.removeContract;
+}
+
+const mapDispatchToProps: DispatchProps = {
+  addContract: contractOperations.addContract,
+  updateContract: contractOperations.updateContract,
+  removeContract: contractOperations.removeContract
+};
+
+export type Props = StateProps & DispatchProps;
+
+export class ContractsPage extends DataTable<Contract, Props> {
+  constructor(props: Props) {
+    super(props);
+    this.addData = this.addData.bind(this);
+    this.updateData = this.updateData.bind(this);
+    this.removeData = this.removeData.bind(this);
+  }
+
+  createNewDataInstance(): Contract {
+    return {
+      tenantId: this.props.tenantId,
+      name: "",
+      maximumMinutesPerDay: null,
+      maximumMinutesPerWeek: null,
+      maximumMinutesPerMonth: null,
+      maximumMinutesPerYear: null 
+    };
+  }
+
+  displayDataRow(data: Contract): JSX.Element[] {
+    return [
+      <Text key={0}>{data.name}</Text>,
+      <Text key={1}>{data.maximumMinutesPerDay}</Text>,
+      <Text key={2}>{data.maximumMinutesPerWeek}</Text>,
+      <Text key={3}>{data.maximumMinutesPerMonth}</Text>,
+      <Text key={4}>{data.maximumMinutesPerYear}</Text>
+    ];
+  }
+  
+  editDataRow(data: Contract): JSX.Element[] {
+    return [
+      <TextInput key={0}
+        name="name"
+        defaultValue={data.name}
+        aria-label="Name"
+        onChange={(value) => data.name = value}
+      />,
+      <OptionalInput key={1}
+        label="Max minutes per day"
+        valueToString={value => value.toString()}
+        defaultValue={data.maximumMinutesPerDay}
+        onChange={value => data.maximumMinutesPerDay = value}
+        isValid={value => /^\d+$/.test(value)}
+        valueMapper={value => parseInt(value)}
+      />,
+      <OptionalInput key={2}
+        label="Max minutes per week"
+        valueToString={value => value.toString()}
+        defaultValue={data.maximumMinutesPerWeek}
+        onChange={value => data.maximumMinutesPerWeek = value}
+        isValid={value => /^\d+$/.test(value)}
+        valueMapper={value => parseInt(value)}
+      />,
+      <OptionalInput key={3}
+        label="Max minutes per month"
+        valueToString={value => value.toString()}
+        defaultValue={data.maximumMinutesPerMonth}
+        onChange={value => data.maximumMinutesPerMonth = value}
+        isValid={value => /^\d+$/.test(value)}
+        valueMapper={value => parseInt(value)}
+      />,
+      <OptionalInput key={4}
+        label="Max minutes per year"
+        valueToString={value => value.toString()}
+        defaultValue={data.maximumMinutesPerYear}
+        onChange={value => data.maximumMinutesPerYear = value}
+        isValid={value => /^\d+$/.test(value)}
+        valueMapper={value => parseInt(value)}
+      />
+    ];
+  }
+  
+  isValid(editedValue: Contract): boolean {
+    return editedValue.name.length > 0;
```

Comment:
I think the frontend validation should prevent sending requests that will fail on the backend. This currently isn't true. For example contract name length is limited to 120 chars, hours must be positive. Is this `editedValue.name.length > 0` condition just a place holder and do you plan on implementing full validation later or should it be implemented now?